syslogish/server.go

```diff
@@ -116,13 +110,6 @@ func (s *Server) processStorage() {
 				labels := messageJSON["kubernetes"].(map[string]interface{})["labels"].(map[string]interface{})
 				// We only want to store deis app log messages
 				if labels != nil && labels["app"] != nil && labels["heritage"] != nil && labels["heritage"].(string) == "deis" {
-					// Get a read lock to ensure the storage adapater pointer can't be nilled by the
-					// configurer in the time between we check if it's nil and the time we invoke .Write()
-					// upon it.
-					s.adapterMutex.RLock()
-					// DONT'T defer unlocking... defered statements are executed when the function returns,
-					// but we are inside an infinite loop here.  If we defer, we would never release the
-					// lock.  Instead, release it manually below.
 					if s.storageAdapter != nil {
```

Comment:
This is the only place I can see that checks any shared state. Do we need to check if the adapter is `nil` at all?